File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int number = 0;
        for(int i=0; i<=nums.size(); i++){
            number ^= i;
        }
        for(int i=0; i<nums.size(); i++){
            number ^= nums[i];
        }
        return number;
    }
};

File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int  n = nums.length;
        long sum = (long) (n*(n+1))/2;
        long curr = 0;

        for(int a: nums){
            curr +=a;
        }
        return (int) (sum - curr);
    }
}